### PR TITLE
[expo-update][ios] Rename swift classes for correct versioning behavior

### PR DESCRIPTION
**Pull request by wschurman**
# Why

Based on my understanding of module transformation, we can't have swift classes that begin with `EX|UM|RCT` if they're imported in objective-c since their names aren't transformed in swift but would be in objective-c. https://github.com/expo/expo/blob/main/tools/src/versioning/ios/transforms/expoModulesTransforms.ts#L11

This PR renames the new swift code relating to code signing in `expo-updates`.

# How

Rename classes and files.

# Test Plan

Run tests.

### CODE REVIEW

File: packages/expo-updates/ios/Tests/CodeSigningSignatureHeaderInfoTests.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
+//  Copyright (c) 2020 650 Industries, Inc. All rights reserved.
+
+import XCTest
+
+@testable import EXUpdates
+
+class CodeSigningSignatureHeaderInfoTests : XCTestCase {
```
Comment (tsapeta) [inline packages/expo-updates/ios/Tests/CodeSigningSignatureHeaderInfoTests.swift:7]:
```suggestion
class CodeSigningSignatureHeaderInfoTests: XCTestCase {
```

The body of the class should also be re-indented 😅 
Besides, could you use `ExpoModulesTestCore` and Quick + Nimble for tests? See tests in `ExpoModulesCore`. I think it's much more convenient way than `XCTest`.

**Review by tsapeta — COMMENTED** [review]
I'm not sure if your understanding is right (it would be great if that's true, we should check this out). So far, for Swift classes that are being used in Objective-C, we're just adding a custom name for Objective-C like this:
```
  @objc(EXUpdatesExpoProjectInformation)
  public class UpdatesExpoProjectInformation: NSObject
```

The versioning script works fine with this approach.

File: packages/expo-updates/ios/Tests/CodeSigningSignatureHeaderInfoTests.swift (hunk 1 above)
Comment (wschurman) [reply packages/expo-updates/ios/Tests/CodeSigningSignatureHeaderInfoTests.swift:7]: I'll add some sort of linting in a follow-up PR so there's no need to review style manually. As for the tests, will also look into it in a follow-up.

For now these are just renames, so to limit the scope of this PR I won't adjust style.

Comment (tsapeta) [reply packages/expo-updates/ios/Tests/CodeSigningSignatureHeaderInfoTests.swift:7]: We actually have SwiftLint set up, but not integrated with Xcode (I think it's just not convenient). My plan was to add this into the code reviewer CI job and even tried that here: #15639 but then it turned out that GitHub rate-limits API tokens when you add too many comments (surprisingly it's sent in only one request).
I'm going to come back to this PR once I'm done with other more urgent stuff.

**Review by tsapeta — APPROVED** [review]
Looks good now. I guess we don't need these changes to land in SDK45, right? If so, I'd wait with merging it until next week.
EDIT: looks like the unit tests are failing on CI 😉

**Comment by wschurman** [comment]
> Looks good now. I guess we don't need these changes to land in SDK45, right? If so, I'd wait with merging it until next week. EDIT: looks like the unit tests are failing on CI 😉

Nope, these changes are required for versioning to work for 45. Also, wee need to backport the current expo-updates to 43 and 44 so these changes are also needed for that (see linked PR).

I'll fix the tests and merge.

**Pull request merged**